src/map/entities/trustentity.cpp

```diff
@@ -108,16 +110,61 @@ void CTrustEntity::Die()
 void CTrustEntity::Spawn()
 {
     //we need to skip CMobEntity's spawn because it calculates stats (and our stats are already calculated)
-    CBattleEntity::Spawn();
+    //CBattleEntity::Spawn();
+    HideName(false);
     ((CCharEntity*)PMaster)->pushPacket(new CTrustSyncPacket((CCharEntity*)PMaster, this));
     luautils::OnMobSpawn(this);
 }
 
 bool CTrustEntity::ValidTarget(CBattleEntity* PInitiator, uint16 targetFlags)
 {
-    if (targetFlags & TARGET_PLAYER && PInitiator->allegiance == allegiance)
+    //check the base flags first before rejecting! now self cures can work :)
+    if (CMobEntity::ValidTarget(PInitiator, targetFlags))
+        return true;
+
+    return !(targetFlags & TARGET_PLAYER && PInitiator->allegiance == allegiance);
+}
+
+void CTrustEntity::OnAbility(CAbilityState& state, action_t& action)
+{
+    auto PAbility = state.GetAbility();
+    auto PTarget = static_cast<CBattleEntity*>(state.GetTarget());
+
+    std::unique_ptr<CBasicPacket> errMsg;
+    if (IsValidTarget(PTarget->targid, PAbility->getValidTarget(), errMsg))
     {
-        return false;
+        if (this != PTarget && distance(this->loc.p, PTarget->loc.p) > PAbility->getRange())
+        {
+            return;
+        }
+        if (battleutils::IsParalyzed(this)) {
```

Comment:
Brace on newline.

Someone else can review the actual logic of the cpp changes.